### PR TITLE
Bc review

### LINKED ISSUE #2: Improve wording on some of the App settings (appspec.jon)
For example:

- [ ] include acceptable values for `timefilter`
- [ ] wording of `altitudecol`, `tempcol` and `headingcol` should clarify that columns __must__ be present in input dataset

### LINKED ISSUE #3: Consistency in settings descriptions between appspecs.json and readme.md
Use the same settings' specifications in the readme and the appspecs.json files.

### PR DESCRIPTION
**Pull request by bcaneco**
Hi guys, 

Submitting the pull request with changes arising from the review. These changes are addressing issues  #2, and #3, as well as other coding modifications and additions (e.g. check on EPSG, CRS, unit tests, etc)

It would be great if you go over the commits and accept/comment them as you see fit.

### CODE REVIEW

**Comment by bcaneco** [comment]
Oh, something else I forgot: I wasn't able to replicate the errors that @callumjclarke  found with the units and the `mutate` function. Running sdk testing with `as.vector()` instances commented out went on as expected, with no errors thrown. 

For the moment I left the `as.vector()` statements active as they were, as I don't want to cause errors in subsequent apps. 

This is no big deal as there are bigger fish to fry atm, but reckon it would be neater to keep the units in all those columns where possible, for completeness and clarity when manipulating the data in the dependent apps.

**Review by lindesaysh — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -2,23 +2,26 @@
...
--   Bin the data to intervals of X minutes
--   Create and append columns containing Universal Transverse Mercator (UTM) data
--   Create and append an *index* column, consisting of the animal's ID and timestamp concatenated
--   Create and append columns containing *speed*, *distance traveled* and/or *time since previous point* data
--   Create and append columns containing specific timestamp data (hour, minute and second, independently)
+-   Thinning the data by filtering one event per defined time window
+-   Create and append columns containing: 
+      - Universal Transverse Mercator (UTM) data
+      - An *index* column, consisting of the animal's ID and timestamp concatenated
+      - *speed*, *distance traveled* and/or *time since previous point* data
```
Comment (lindesaysh) [inline README.md]: is it time since previous point or time to next point?

File: README.md [hunk 2]
```diff
@@ -45,46 +48,45 @@ Move2 location object
 
 `Filter by Time Interval` (integer): The length of interval, in minutes, to which to filter data. Must be between 0 and 60. Setting this equal to 0 means no filtering will take place. Defaults to 0 (no filtering). The function `move2::mt_filter_per_interval` is used for this filtering with the parameter `criterion = first`.
 
-`Upper Threshold for Movement Speed` (numeric): The highest acceptable movement speed for all IDs (units: km/h). Any location with a speed exceeding this value will be removed as an outlier. 
+`Bind additional timestamp columns` (logical): Determines whether to append specific timestamp data (*hour,* *minute,* *second,* *hourmin,* and *yearmonthday*) to the output. Default: `TRUE`.
 
-`Bind additional timestamp columns` (logical): Determines whether to append specific timestamp data (*hour,* *minute,* *second,* *hourmin,* and *yearmonthday*) to the output. 
+`Bind time difference` (logical): Determines whether to append a column with the *time difference* since the previous location (units: hours). Default: `TRUE`.
```
Comment (lindesaysh) [inline README.md]: same comment as before

File: README.md [hunk 3]
```diff
@@ -45,46 +48,45 @@ Move2 location object
...
 
-    1.  That the named column is **contained within the input dataset**
-    2.  That the column name has been **spelled with complete accuracy**
+    1.  the named column is **contained within the input dataset**
+    2.  the column name has been **spelled with complete accuracy**
 
     or an error will be returned and processing will stop. Check the logs for the list of column names in the input dataset.
 
--   Any `column` settings can be bugged by a column name with two (and, sometimes, one) periods: for example, 'altitude.col.xyz' could not be recognized. To solve this bug, any column containing a period must be duplicated (instead of renamed) with the new name `altutide/temperature/heading`.
+-   Any `column` settings can be bugged by a column name with two (and, sometimes, one) periods: for example, 'altitude.col.xyz' could not be recognized. To solve this bug, any column containing a period must be duplicated (instead of renamed) with the new name `altitude/temperature/heading`.
```
Comment (lindesaysh) [inline README.md:84]: is this something the user has to do?  how would they go about changing a column name of something coming from moveapps?

**Review by lindesaysh — COMMENTED** [review]
i've left a few comments.  not sure how this works exactly!

**Comment by bcaneco** [comment]
Thanks Lindesay! I'll go over your comments and do required amendments.

File: README.md (hunk 1 above)
Comment (bcaneco) [reply README.md]: Yep, you're absolutely right! Will amend this and commit again

File: README.md (hunk 2 above)
Comment (bcaneco) [reply README.md]: same as before

File: README.md (hunk 3 above)
Comment (bcaneco) [reply README.md:84]: Yeah, I'm not sure about that statement either. My understanding is that the code is solving the issue automatically, i.e. it duplicates the column when it's name contains a period.  @callumjclarke, can you confirm please?

**Review by lindesaysh — APPROVED** [review]

**Review by callumjclarke — APPROVED** [review]

**Pull request merged**